### PR TITLE
Scoped identity at product alert email

### LINKED ISSUE #28968: [Issue] Scoped identity at product alert email
This issue is automatically created based on existing pull request: magento/magento2#26534: Scoped identity at product alert email

---------

### Summary (*)
Magento 2.4-develop
Price/stock alert email was always sended with identity from default store. I fixed that identity sets from store where alert is subscribed

### Examples (*)
1. Subscribe to price/stock alert on other store than default
2. When notification is sended to customer, as a sender will be contact from scoped store

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by maqlec**
### Description (*)

Price/stock alert email was always sended with identity from default store. I fixed that identity sets from store where alert is subscribed

### Manual testing scenarios (*)

1. Subscribe to price/stock alert on other store than default
2. When notification is sended to customer, as a sender will be contact from scoped store

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#28968: Scoped identity at product alert email

### CODE REVIEW

File: app/code/Magento/ProductAlert/Model/Email.php [hunk 1]
```diff
@@ -378,12 +378,12 @@ public function send()
...
                 'alertGrid' => $alertGrid,
             ]
-        )->setFrom(
+        )->setFromByScope(
             $this->_scopeConfig->getValue(
                 self::XML_PATH_EMAIL_IDENTITY,
                 ScopeInterface::SCOPE_STORE,
                 $storeId
-            )
+            ), $storeId
```
Comment (Stepa4man) [inline app/code/Magento/ProductAlert/Model/Email.php]: Please move this argument declaration to the new line

**Review by Stepa4man — CHANGES_REQUESTED** [review]
Hi @maqlec 
Thanks for your contribution

Please check the test failures.

File: app/code/Magento/ProductAlert/Model/Email.php (hunk 1 above)
Comment (maqlec) [reply app/code/Magento/ProductAlert/Model/Email.php]: thanks, its done.

**Comment by maqlec** [comment]
@Stepa4man failing tests are unrelated to changes. Is there a way to retest only one of them?

**Comment by maqlec** [comment]
> Hi @maqlec
> Thanks for your contribution
> 
> Please check the test failures.

all checks have passed :)

**Review by Stepa4man — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Stepa4man, thank you for the review.
[ENGCOM-7506](https://jira.corp.magento.com/browse/ENGCOM-7506) has been created to process this Pull Request
:eight_spoked_asterisk: @Stepa4man, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@maqlec thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by engcom-Delta — CHANGES_REQUESTED** [review]
@maqlec Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests.
Seems like new behavior not covered by existing `dev/tests/integration/testsuite/Magento/ProductAlert/Model/EmailTest.php` test

**Comment by maqlec** [comment]
@magento run Integration Tests

**Comment by maqlec** [comment]
@engcom-Delta failed tests are not related to changes

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by Stepa4man** [comment]
@magento run all test

**Comment by Stepa4man** [comment]
@magento run all tests

**Comment by Stepa4man** [comment]
Hi @maqlec,

Please add to all changed (created) php files
```php
    declare(strict_types=1);
```
You can read more [here](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html#1-basic-programming-principles)

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by Stepa4man — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Stepa4man, thank you for the review.
[ENGCOM-7506](https://jira.corp.magento.com/browse/ENGCOM-7506) has been created to process this Pull Request
:eight_spoked_asterisk: @Stepa4man, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Manual testing scenario:**

- Create second store view
 image
- Change Sender Email for Second Store View
 image
- Configure price/stock alert
 image
- Set Display Out of Stock Products=Yes
 image
- Create two simple products, one of them is Out of Stock
- Create customer
- Change view to Second Store view on frontend
- For product In Stock, click "Notify me when the price drops" for subscribe to price alert on frontend
 image
- For product Out of Stock, click "Notify me when this product is in stock" for subscribe to price alert on frontend
 image
- Go to admin panel
- Set lower price for product In Stock
- Set Stock Status=In Stock for product that was Out of Stock
- Change local date to next day
- Run `cron:run` twice
- Check customer mail

:x: **Before:**
Sender Email is from default view
#26534MainStock
#26534MainPRice


:heavy_check_mark: **After:**
Sender Email is from view where customer was subscribed for alerts(from Second Store View)
#26534PRStock
#26534PRPrice

**Comment by engcom-Delta** [comment]
@magento create issue

**Pull request merged**